utils/ranges.hh

```diff
@@ -25,7 +25,9 @@
 
 namespace ranges {
 
-template <std::ranges::range Container, std::ranges::range Range>
+template <std::ranges::range Container, typename Range>
```

Comment:
Maybe it would be better to use '#if' to use the original version in gcc and the other version in clang, with a comment saying (like you did) why this version exists. Sometime in the future, somebody will remove the extra case. I'm worried that with this change, nobody will ever remember to add the concept back.

Reply:
Good idea, will change.